### PR TITLE
Don't build eBPF probes for kernels < 4.14

### PR DESCRIPTION
**Pull request by gnosek**

### CODE REVIEW

File: build-probe-binaries [hunk 1]
```diff
@@ -400,6 +400,20 @@ function build_probe_impl {
...
+    if [ "$KERNEL_MAJOR" -lt 4 ]
+    then
+        echo "Kernel $KERNEL_RELEASE too old to support eBPF (need at least 4.14), skipping eBPF build"
+        return
+    elif [ "$KERNEL_MAJOR" -eq 4 -a "$KERNEL_MINOR" -lt 14 ]
+    then
+        echo "Kernel $KERNEL_RELEASE too old to support eBPF (need at least 4.14), skipping eBPF build"
+        return
+    fi
+
```
Comment (iurly) [inline build-probe-binaries]: Could you please use tabs instead of spaces, in order to keep consistency with the existing code?

**Review by iurly — CHANGES_REQUESTED** [review]
LGTM, thanks!

**Comment by irozzo-1A** [comment]
@gnosek Just for personal knowledge, why do we need 4.14 at least? Is it for BPF support for [skbs on sockets](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md#main-features)?

**Comment by gnosek** [comment]
@irozzo-1A, we'd have to explore the history of this line: https://github.com/draios/agent-libs/blob/dev/driver/bpf/quirks.h#L14-L16

**Review by irozzo-1A — DISMISSED** [review]
LGTM
fair enough @gnosek I'll take a look

**Comment by irozzo-1A** [comment]
I did not take the time to understand the exact reason, but the constraint comes from here:
https://github.com/draios/agent-libs/commit/19317d0028e14264fbd3908e74eed439123b73bb#diff-e8ed0901afbcbaf7c1a402e0d209afa2743604a98ff796ee5bbfa23fced4c4bf

**Comment by gnosek** [comment]
Yeah, that's about where I gave up too. You're in ancient history territory now ;)

BTW, I just did a successful builder pass for AmazonLinux 1, with 81 kmods and 55 eBPF probes built.

**Review by iurly — APPROVED** [review]
Ship it! (please...) :-)

**Pull request merged**